src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -560,6 +560,38 @@ static SpanSource ThrowUnhandledNodeType(TrieNode item)
             return length == 32 ? rlpStream.DecodeKeccak() : null;
         }
 
+        /// Gets child hash or the RLP of the node in case it's an inline.
+        public byte[]? GetChildHashOrRlp(int i)
+        {
+            SpanSource rlp = _rlp;
+            if (rlp.IsNull)
+            {
+                return null;
+            }
+
+            ValueRlpStream rlpStream = new(rlp);
+            SeekChild(ref rlpStream, i);
+
+            int prefix = rlpStream.PeekByte();
+
+            // If it's a hash (32 bytes), decode and return it
+            if (prefix == 160)
+            {
+                return rlpStream.DecodeKeccak().Bytes.ToArray();
+            }
+
+            int prefixValue = rlpStream.PeekByte();
+            if (prefixValue < 192)
+            {
+                return null;
+            }
+            else
+            {
+                // If it's an RLP list (inline node), return the full item as a byte array
+                return rlpStream.PeekNextItem().ToArray();
```

Comment:
The only issue I have here is we have RlpStream here and we allocate an array, then later we put that Array into another RlpStream and unpack and allocate another array:

https://github.com/NethermindEth/nethermind/pull/9367/files#diff-6f44b90238bac9456c438d2b0bb575e3df9bc3f4c03ef7c03653c9ba77555314R192-R196

could we do unpacking here and return the final array here?